File: piwigo/Image/Parameters/Cells/EditImageTextViewTableViewCell.m
//
//  EditImageTextViewTableViewCell.m
//  piwigo
//
//  Created by Spencer Baker on 2/8/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//

#import "EditImageTextViewTableViewCell.h"
#import "Model.h"

@interface EditImageTextViewTableViewCell()

@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation EditImageTextViewTableViewCell

- (void)awakeFromNib {
    // Initialization code
    [super awakeFromNib];
    
    self.label.text = NSLocalizedString(@"editImageDetails_description", @"Description");
    self.label.font = [UIFont piwigoFontNormal];

    self.textView.font = [UIFont piwigoFontNormal];
    self.textView.keyboardType = UIKeyboardTypeDefault;
    self.textView.autocapitalizationType = UITextAutocapitalizationTypeSentences;
    self.textView.autocorrectionType = UITextAutocorrectionTypeYes;
    self.textView.returnKeyType = UIReturnKeyDefault;
    self.textView.layer.cornerRadius = 5.0;
}

-(void)setComment:(NSString *)imageDetail inColor:(UIColor *)color
{
    // Cell background
    self.backgroundColor = [UIColor piwigoColorBackground];
    
    // Cell label
    self.label.textColor = [UIColor piwigoColorRightLabel];

    // Cell text view
    self.textView.text = imageDetail;
    self.textView.textColor = color;
    self.textView.backgroundColor = [UIColor piwigoColorBackground];
    self.textView.keyboardAppearance = [Model sharedInstance].isDarkPaletteActive ? UIKeyboardAppearanceDark : UIKeyboardAppearanceDefault;
}

-(void)prepareForReuse
{
    [super prepareForReuse];

    self.textView.delegate = nil;
    self.textView.text = @"";
}

@end

File: piwigo/Image/Parameters/Cells/EditImageTextViewTableViewCell.swift
//
//  EditImageTextViewTableViewCell.swift
//  piwigo
//
//  Created by Spencer Baker on 2/8/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//
//  Converted to Swift 5.3 by Eddy Lelièvre-Berna on 29/08/2021.
//

import UIKit

class EditImageTextViewTableViewCell: UITableViewCell {
    
    @IBOutlet weak var textView: UITextView!
    @IBOutlet private weak var label: UILabel!

    override func awakeFromNib() {
        // Initialization code
        super.awakeFromNib()

        label.text = NSLocalizedString("editImageDetails_description", comment: "Description")
        textView.layer.cornerRadius = 5.0
    }

    func config(withText imageDetail: NSAttributedString?, inColor color: UIColor?) {
        // Cell background
        backgroundColor = PwgColor.background

        // Cell label
        label.textColor = PwgColor.leftLabel

        // Cell text view
        textView.attributedText = imageDetail
        textView.textColor = color
        textView.backgroundColor = PwgColor.background
        textView.keyboardAppearance = AppVars.shared.isDarkPaletteActive ? .dark : .default
    }

    override func prepareForReuse() {
        super.prepareForReuse()

        textView.delegate = nil
        textView.text = ""
    }
}
